lib/internal/stream_base_commons.js

```diff
@@ -84,8 +84,8 @@ function onWriteComplete(status) {
   }
 
   if (status < 0) {
-    const ex = errnoException(status, 'write', this.error);
-    stream.destroy(ex, this.callback);
+    if (typeof this.callback === 'function')
+      this.callback(errnoException(status, 'write', this.error));
```

Comment:
@addaleax: This is the only part that should be relevant for the `stdio-close` errors.